src/m365/aad/commands/user/user-add.ts

```diff
@@ -0,0 +1,239 @@
+  lastName?: string;
+  forceChangePasswordNextSignIn: boolean;
+  forceChangePasswordNextSignInWithMfa: boolean;
+  usageLocation?: string;
+  jobTitle?: string;
+  companyName?: string;
+  department?: string;
+  preferredLanguage?: string;
+  managerUserId?: string;
+  managerUserName?: string;
+}
+
+class AadUserAddCommand extends GraphCommand {
+  public get name(): string {
+    return commands.USER_ADD;
+  }
+
+  public get description(): string {
+    return 'Creates a new user';
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+    this.#initTypes();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        accountEnabled: typeof args.options.accountEnabled !== 'undefined',
+        mailNickname: typeof args.options.mailNickname !== 'undefined',
+        password: typeof args.options.password !== 'undefined',
+        firstName: typeof args.options.firstName !== 'undefined',
+        lastName: typeof args.options.lastName !== 'undefined',
+        forceChangePasswordNextSignIn: !!args.options.forceChangePasswordNextSignIn,
+        forceChangePasswordNextSignInWithMfa: !!args.options.forceChangePasswordNextSignInWithMfa,
+        usageLocation: typeof args.options.usageLocation !== 'undefined',
+        jobTitle: typeof args.options.jobTitle !== 'undefined',
+        companyName: typeof args.options.companyName !== 'undefined',
+        department: typeof args.options.department !== 'undefined',
+        preferredLanguage: typeof args.options.preferredLanguage !== 'undefined',
+        managerUserId: typeof args.options.managerUserId !== 'undefined',
+        managerUserName: typeof args.options.managerUserName !== 'undefined'
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '--displayName <displayName>'
+      },
+      {
+        option: '--userName <userName>'
+      },
+      {
+        option: '--accountEnabled [accountEnabled]'
+      },
+      {
+        option: '--mailNickname [mailNickname]'
+      },
+      {
+        option: '--password [password]'
+      },
+      {
+        option: '--firstName [firstName]'
+      },
+      {
+        option: '--lastName [lastName]'
+      },
+      {
+        option: '--forceChangePasswordNextSignIn'
+      },
+      {
+        option: '--forceChangePasswordNextSignInWithMfa'
+      },
+      {
+        option: '--usageLocation [usageLocation]'
+      },
+      {
+        option: '--jobTitle [jobTitle]'
+      },
+      {
+        option: '--companyName [companyName]'
+      },
+      {
+        option: '--department [department]'
+      },
+      {
+        option: '--preferredLanguage [preferredLanguage]'
+      },
+      {
+        option: '--managerUserId [managerUserId]'
+      },
+      {
+        option: '--managerUserName [managerUserName]'
+      }
```

Comment:
Looking at the response object I feel like we should add another property `officeLocation`. Could you add that one as well please? I've added it in the specs as well.